ansible/roles/test/files/helpers/bfd_responder.py

```diff
@@ -0,0 +1,184 @@
+import binascii
+import socket
+import struct
+import select
+import json
+import argparse
+import os.path
+from fcntl import ioctl
+import logging
+logging.getLogger("scapy.runtime").setLevel(logging.ERROR)
+import scapy.all as scapy2
+scapy2.conf.use_pcap=True
+from  scapy.contrib.bfd import BFD
+def hexdump(data):
+    print (" ".join("%02x" % ord(d) for d in data))
+
+def get_if(iff, cmd):
+    s = socket.socket()
+    ifreq = ioctl(s, cmd, struct.pack("16s16x",iff))
+    s.close()
+
+    return ifreq
+
+def get_mac(iff):
+    SIOCGIFHWADDR = 0x8927          # Get hardware address
+    return get_if(iff, SIOCGIFHWADDR)[18:24]
+
+
+class Interface(object):
+
+    def __init__(self, iface):
+        self.iface = iface
+        self.socket = None
+        self.mac_address = get_mac(iface)
+
+    def __del__(self):
+        if self.socket:
+            self.socket.close()
+
+    def bind(self):
+        self.socket = scapy2.conf.L2listen(iface=self.iface, filter="udp port 4784")
+
+    def handler(self):
+        return self.socket
+
+    def recv(self):
+        sniffed = self.socket.recv()
+        pkt = sniffed[0]
+        str_pkt = str(pkt).encode("HEX")
+        binpkt = binascii.unhexlify(str_pkt)
+        return binpkt
+
+    def send(self, data):
+        scapy2.sendp(data, iface=self.iface)
+
+    def mac(self):
+        return self.mac_address
+
+    def name(self):
+        return self.iface
+
+
+class Poller(object):
+    def __init__(self, interfaces, responder):
+        self.responder = responder
+        self.mapping = {}
+        for interface in interfaces:
+            self.mapping[interface.handler()] = interface
+
+    def poll(self):
+        handlers = self.mapping.keys()
+        while True:
+            (rdlist, _, _) = select.select(handlers, [], [])
+            for handler in rdlist:
+                self.responder.action(self.mapping[handler])
+
+
+class BFDResponder(object):
+    def __init__(self, sessions):
+        self.sessions = sessions
+        return
+
+    def action(self, interface):
+        data = interface.recv()
+        mac_src, mac_dst, ip_src, ip_dst,  bfd_remote_disc, bfd_state = self.extract_bfd_info(data)
+        if ip_dst not in self.sessions:
+            return
+        session = self.sessions[ip_dst]
+
+        if bfd_state== 3L:
+            interface.send(session["pkt"])
+            return
+
+        if bfd_state == 2L:
+            return
+        session = self.sessions[ip_dst]
+        session["other_disc"] = bfd_remote_disc
+        bfd_pkt_init = self.craft_bfd_packet( session, data, mac_src, mac_dst, ip_src, ip_dst, bfd_remote_disc, 2L )
+        print "sending INIT", bfd_pkt_init
```

Comment:
This print is incompatible with python3.

Reply:
fixed